presto-main/src/main/java/com/facebook/presto/execution/scheduler/SqlQueryScheduler.java

```diff
@@ -287,8 +287,8 @@ public BasicStageExecutionStats getBasicStageStats()
     public StageInfo getStageInfo()
     {
         Map<StageId, StageExecutionInfo> stageInfos = stageExecutions.values().stream()
-                .map(stageExecutionInfo -> stageExecutionInfo.getStageExecution().getStageExecutionInfo())
```

Comment:
I'm not entirely sure if this change is related, but it looks fine so we could keep it as is.

Reply:
it's related because since we removed the id from StageExecutionInfo, we can't use it for the key in our map.  We need to get the key from the StageExecution.  (it's a bit confusing because the argument to the lambda was also called stageExecutionInfo even though it's a StageExecutionAndScheduler)